packages/shared/ReactServerContext.js

```diff
@@ -0,0 +1,105 @@
+/**
```

Comment:
This shouldn't be shared since it contains module level state that needs to be shared. This one should be in `/react/`

The other shared case should import createServerContext from `react`. That way we don't ship the implementation twice.